### PR TITLE
Added code comments to the python interface of ek and lb boundaries a…

### LINKED ISSUE #1117: lbboundaries.py lb.pyx ekboundaries.py electrokinetics.pyx

### PR DESCRIPTION
**Pull request by itischler**
Fixes #1117

Description of changes:
 - Documentation comments on lbboundary.py
 - Documentation comments on ekboundary.py
 - Documentation comments on electokinetics.pyx

PR Checklist
------------
 - [ ] Tests?
   - [ ] Interface
   - [ ] Core 
 - [ ] Docs?

### CODE REVIEW

File: src/python/espressomd/ekboundaries.py [hunk 1]
```diff
@@ -4,9 +4,17 @@
 
 @script_interface_register
 class EKBoundaries(ScriptInterfaceHelper):
+    """
+    Creates a set of electrokinetics boundaries.
```
Comment (KaiSzuttor) [inline src/python/espressomd/ekboundaries.py:8]: newline after that line

File: src/python/espressomd/ekboundaries.py [hunk 2]
```diff
@@ -4,9 +4,17 @@
...
+    """
     _so_name = "LBBoundaries::LBBoundaries"
 
     def add(self, *args, **kwargs):
+        """
+        Adds a boundary to the set.
+        Either a valid boundary is an argument,
+        or a valid set of parameters to create a boundary.
```
Comment (KaiSzuttor) [inline src/python/espressomd/ekboundaries.py:17]: newline after that line

File: src/python/espressomd/ekboundaries.py [hunk 3]
```diff
@@ -19,9 +27,20 @@ def add(self, *args, **kwargs):
         return ekboundary
 
     def remove(self, ekboundary):
+        """
+        Removes a boundary from the set.
+        
+        Parameters
+        ----------
+        ekboundary : :obj: `EKBoundary`
+                     The boundary to be removed from the set.
```
Comment (KaiSzuttor) [inline src/python/espressomd/ekboundaries.py]: newline after that line

File: src/python/espressomd/ekboundaries.py [hunk 4]
```diff
@@ -19,9 +27,20 @@ def add(self, *args, **kwargs):
...
+        """
         self.call_method("remove", ekboundary=ekboundary)
 
 
 @script_interface_register
 class EKBoundary(ScriptInterfaceHelper):
+    """
+    Creates a EK boundary.
```
Comment (KaiSzuttor) [inline src/python/espressomd/ekboundaries.py:49]: newline after that line

File: src/python/espressomd/electrokinetics.pyx [hunk 5]
```diff
@@ -4,9 +4,13 @@ from .lb cimport HydrodynamicInteraction
 from .ekboundaries import EKBoundary
 from . import utils
 import numpy as np
+from espressomd.utils import is_valid_type
 
 IF ELECTROKINETICS:
     cdef class Electrokinetics(HydrodynamicInteraction):
+        """
+        Creates the electrokinetic method using the GPU unit.
```
Comment (KaiSzuttor) [inline src/python/espressomd/electrokinetics.pyx:12]: newline after that line

File: src/python/espressomd/electrokinetics.pyx [hunk 6]
```diff
@@ -18,6 +22,9 @@ IF ELECTROKINETICS:
 
 
         def validate_params(self):
+            """
+            Checks if the parameters for "stencil" and "fluid_coupling" are valid.
```
Comment (KaiSzuttor) [inline src/python/espressomd/electrokinetics.pyx:28]: newline after that line

File: src/python/espressomd/electrokinetics.pyx [hunk 7]
```diff
@@ -28,12 +35,21 @@ IF ELECTROKINETICS:
 
 
         def valid_keys(self):
-            return "agrid", "lb_density", "viscosity", "friction", "bulk_viscosity", "gamma_even", "gamma_odd", "T", "bjerrum_length", "stencil", "advection", "fluid_coupling"
+            """
+            Retruns the valid options used for the electrokinetic method.
```
Comment (KaiSzuttor) [inline src/python/espressomd/electrokinetics.pyx]: typo returns

newline after that line

File: src/python/espressomd/electrokinetics.pyx [hunk 8]
```diff
@@ -28,12 +35,21 @@ IF ELECTROKINETICS:
...
+            """
+            return "agrid", "lb_density", "viscosity", "friction", "bulk_viscosity", "gamma_even", "gamma_odd", "T", "prefactor", "stencil", "advection", "fluid_coupling"
 
         def required_keys(self):
-            return ["agrid", "lb_density", "viscosity", "friction", "T", "bjerrum_length"]
+            """
+            Retruns the nessesary options to initialize the electokinetic method.
```
Comment (KaiSzuttor) [inline src/python/espressomd/electrokinetics.pyx]: typo returns

newline after that line

File: src/python/espressomd/electrokinetics.pyx [hunk 9]
```diff
@@ -28,12 +35,21 @@ IF ELECTROKINETICS:
...
+            """
+            return ["agrid", "lb_density", "viscosity", "friction", "T", "prefactor"]
 
         def default_params(self):
+            """
+            Retruns the default paramters.
```
Comment (KaiSzuttor) [inline src/python/espressomd/electrokinetics.pyx]: typo returns

newline after that line

File: src/python/espressomd/electrokinetics.pyx [hunk 10]
```diff
@@ -94,22 +110,37 @@ IF ELECTROKINETICS:
...
         def set_density(self, species=None, density=None, node=None):
+            """
+            Sets the density of a species at a speciffic node.
+            If no node is given the density will be set global for the species
+            
+            Parameters
+            ----------
+            species : :obj: `integer`
+                      species for which the density will apply.
+
```
Comment (KaiSzuttor) [inline src/python/espressomd/electrokinetics.pyx]: no newlines between parameters necessary

File: src/python/espressomd/electrokinetics.pyx [hunk 11]
```diff
@@ -146,28 +196,89 @@ IF ELECTROKINETICS:
...
+            Parameters
+            ----------
+            species : :obj: `integer`
+                      Species to be initialized.
+            """
             self.species_list.append(species)
 
         def get_params(self):
+            """
+            Prints out the parameters of the electrokinetic system.
```
Comment (KaiSzuttor) [inline src/python/espressomd/electrokinetics.pyx:227]: please add newlines at the end of every docstring

File: src/python/espressomd/electrokinetics.pyx [hunk 12]
```diff
@@ -255,12 +366,21 @@ IF ELECTROKINETICS:
                     raise KeyError("%s is not a vaild key" % k)
 
         def valid_keys(self):
+            """
+            Returns the valid keys for the species
```
Comment (KaiSzuttor) [inline src/python/espressomd/electrokinetics.pyx]: please end with a period

File: src/python/espressomd/electrokinetics.pyx [hunk 13]
```diff
@@ -281,13 +401,32 @@ IF ELECTROKINETICS:
...
             self._params.update(self._get_params_from_es_core())
             return self._params
 
         def print_vtk_density(self, path):
+            """
+            Writes the species density into a vtk-file.
+            
+            Parameters
+            ----------
+            path : :obj: `string`
```
Comment (KaiSzuttor) [inline src/python/espressomd/electrokinetics.pyx]: no space between obj and the ticks

File: src/python/espressomd/lbboundaries.py [hunk 14]
```diff
@@ -21,10 +29,21 @@ def add(self, *args, **kwargs):
             return lbboundary
 
         def remove(self, lbboundary):
+            """
+            Removes a boundary from the set.
+            
+            Parameters
+            ----------
+            lbboundary : :obj: `LBBoundary`
```
Comment (KaiSzuttor) [inline src/python/espressomd/lbboundaries.py]: is a link generated here? please check, if not you have to provide the full path

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]

**Comment by KaiSzuttor** [comment]
please pull in the current python branch, your base branch seems *very* old

**Comment by fweik** [comment]
@KaiSzuttor please merge if you're happy, your points seem to by mostly addressed.

**Pull request merged**